src/service/cia301/co_csdo.c

```diff
@@ -35,10 +41,35 @@ static CO_ERR COCSdoUploadSegmented        (CO_CSDO *csdo);
 static CO_ERR COCSdoInitDownloadSegmented  (CO_CSDO *csdo);
 static CO_ERR COCSdoDownloadSegmented      (CO_CSDO *csdo);
 static CO_ERR COCSdoFinishDownloadSegmented(CO_CSDO *csdo);
+
+static CO_ERR COCSdoInitUploadBlock        (CO_CSDO *csdo);
+static CO_ERR COCSdoUploadSubBlock         (CO_CSDO *csdo);
+static CO_ERR COCSdoInitDownloadBlock      (CO_CSDO *csdo);
+static CO_ERR COCSdoDownloadSubBlock_Request    (CO_CSDO *csdo);
+static CO_ERR COCSdoDownloadSubBlock       (CO_CSDO *csdo);
+static CO_ERR COCSdoFinishDownloadBlock    (CO_CSDO *csdo);
+
 static void   COCSdoAbort                  (CO_CSDO *csdo, uint32_t err);
 static void   COCSdoTransferFinalize       (CO_CSDO *csdo);
 static void   COCSdoTimeout                (void *parg);
 
+CO_ERR COCSdoRequestDownloadFull(CO_CSDO *csdo,
+                             uint32_t key,
+                             uint8_t *buffer,
+                             uint32_t size,
+                             CO_CSDO_CALLBACK_T callback,
+                             uint32_t timeout,
+                             blockTransfer_t* block);
+
+// cbt TODO: change bool block to structure like for download
```

Comment:
This comment is really confusingly worded but if you are the one who is going to work on it after this I guess you don't have to change it at all.